Font sizing, site width, dropdown nav items

daviddarnes opened the pull request:
- Adjust font sizing
- Adjust site width size
- Add dropdown nav to header nav

odow commented on assets/styles.scss:
```diff
@@ -0,0 +1,96 @@
+  text-align: center;
+  padding: .75rem 1.25rem;
+  margin-bottom: 1rem;
+  background-color: $codeBackgroundColour;
```
Hmm, so I tried changing the color here to `"red"` and `#f00` without success. Is it possible to make the banner background orange (or dark-grey) to stand out? What am I missing?

daviddarnes replied:
@odow strange you couldn't change it, guessing you were using the correct build commands? I'll change it to a tint of orange.

mlubin commented on _layouts/page-with-notification.html:
```diff
@@ -0,0 +1,31 @@
+---
+
+{%- capture notification %}
+Testing
```
Could this be a value that's defined in the config file?

daviddarnes replied:
@mlubin sure! Will shift it over

mlubin commented on _config.yml:
```diff
@@ -72,12 +72,13 @@ collections:
+      path: ""
+      types: "pages"
+    values:
+      layout: "page-with-notification"
```
To follow up on @odow's question, could you add a short comment on how to switch the notification bar on and off?

odow replied:
I figured it out. You just comment out this section.

daviddarnes replied:
@mlubin @odow I've changed this now so you can just comment out the notification in the config file to turn it off. No need for that default stuff anymore

mlubin commented on _includes/nav-header.html:
```diff
@@ -0,0 +1,82 @@
+  {% include nav-default.html %}
+{% endif %}
+
+<script type="text/javascript">
```
Mostly out of curiosity, what is this javascript doing? A short comment would suffice.

daviddarnes replied:
@mlubin the js does the nav toggle on mobile, hiding the nav on mobile and listening for clicks on the nav button. Jekyll lets you overwrite includes, I needed to do it to add the dropdown. Alembic has this same js in it's `nav-header.html`, so I had to copy it in too 🙂

mlubin commented:
There's an issue with drop-down menu expanding and unexpanding in response to a window resize:
![jump dev](https://user-images.githubusercontent.com/1733683/72198267-3b1f4600-33f9-11ea-88b0-be35353381b1.gif)
It seems to be a new issue, doesn't happen with the current jump.dev.

daviddarnes commented:
@mlubin I've put in a fix for the flickering nav issue. Also changed the banner colour and moved the message to the config. Now that the notification is in the config I was able to use it as a flag, meaning if there's no notification there's no banner. Also meant that we didn't need the extra layout file and we could overwrite the regular page layout. As an extra bit I've allowed for markdown use in the notification 👍

daviddarnes commented:
@odow just pushed a fix to sort that out 👍

daviddarnes commented:
@odow I've just done some tweaking so the notification banner appears on all pages. Is that what you wanted?

mlubin commented:
@daviddarnes, thanks for the quick fixes. I'd like to try out the new changes but I have a crazy schedule this week. I'll plan to do it over the weekend.

mlubin commented:
Looks great, thanks!!
